backend/matchmaker_service.py

```diff
@@ -0,0 +1,170 @@
+import firebase_admin
+from firebase_admin import credentials, firestore
+from game import Game
+from game_engine.common import GameOptions, GameSchedule, STV_I18N_TABLE, ISODayOfWeek, log_message
+from game_engine.database import Database
+from game_engine.engine import Engine
+from game_engine.firestore import FirestoreDB
+from game_engine.matchmaker import MatchMakerInterface
+from test_game import MockDatabase, MockPlayEngine
+from google.cloud.firestore_v1.document import DocumentSnapshot
+import time
+import threading
+import datetime
+from game_engine import events
+from multiprocessing import Process
+import multiprocessing
+
+_FIRESTORE_PROD_CONF_JSON_PATH = ''
+_TEST_FIRESTORE_INSTANCE_JSON_PATH = '../firebase/stv-game-db-test-4c0ec2310b2e.json'
+json_config_path = _TEST_FIRESTORE_INSTANCE_JSON_PATH
+_AMAZON_SQS_PROD_CONF_JSON_PATH = '../amazon/stopthevirus.fifo.json'
+_TEST_AMAZON_SQS_CONFIG_PATH = '../amazon/stopthevirus.fifo.json'
+_TEST_TWILIO_SMS_CONFIG_PATH = '../twilio/stv-twilio.json'
+
+
+
+class MatchmakerService:
+    # Handles scheduling and communication with other services for starting games
+    # TDOO(David): Add function to run all games that are supposed to be running at start(in MVP/test)
+    def __init__(self, matchmaker: MatchMakerInterface, gamedb: Database, json_config_path=json_config_path, region="US", min_players=5, is_mvp=True):
+        self._matchmaker = matchmaker
+        self._gamedb = gamedb
+        self._min_players = min_players
+        self._region = region
+        self._is_mvp = is_mvp
+        self._stop = threading.Event()
+        self._daemon_started = False
+
+    def play_game(self, game: Game, players:list, game_dict:dict, is_test=True):
+        log_message("Starting a game", game_id=game_dict.get("id"), additional_tags=game_dict)
+
+        if is_test:
+            database = MockDatabase()

```

Comment:
(optional) consider making database and engine arguments to the play_game method rather than including the is_test argument. this way, the dependencies can be injected into the method call. in our unit test we'd pass 'MockDatabase' and 'MockPlayEngine' instances.

Reply:
That sounds like a good idea but the issue is that new DB and engine instances need to be created for each game, because the other functions we call get the game_id and other parameters from them. With the test MockDatabase it doesn't really matter but for the actual implementation this is true. I could move this logic up to _start_game but it wouldn't really change much.